scripts/pull_translations.rb

```diff
@@ -0,0 +1,167 @@
+      puts "Updating translation to %s" % translation if show_info
+      translation.content = translations_hash[key].translated_string
+    end
+  end
+  File.write(filename, strings_doc.to_xml)
+end
+
+def add_new_translations(filename, translations_hash, show_info)
+  puts '▸ Adding new translations'
+  strings_doc = File.open(filename) { |f| Nokogiri::XML(f) }
+  translation_nodes = strings_doc.xpath('//string')
+  existing_strings = strings_doc.xpath('//string/@name').to_a.to_set(&:text)
+  new_strings = translations_hash.keys.to_set - existing_strings
+  # Grab a new line node to use for formatting
+  new_line_node = translation_nodes.first.next.dup
+  # Move the last new line to the end
+  last_newline_node = translation_nodes.last.next.unlink
+  new_strings.each do |string|
+    translation = translations_hash[string]
+    unless translation.description_comment.nil?
+      strings_doc.root.add_child(new_line_node.dup)
+      strings_doc.root.add_child(translation.description_comment)
+    end
+    strings_doc.root.add_child(new_line_node.dup)
+    strings_doc.root.add_child(translation.string_node)
+    puts "Added translation for %s" % translation.string_node if show_info
+  end
+  strings_doc.root.add_child(last_newline_node)
+  File.write(filename, strings_doc.to_xml)
+end
+
+def update_translations(default_strings_file, supported_languages, show_info=true)
+  puts '▸ Building set of locally added strings'
+  default_strings_doc = File.open(default_strings_file) { |f| Nokogiri::XML(f) }
+  included_strings = default_strings_doc.xpath('//string/@name')
+                                        .to_a.to_set(&:text)
+  puts '▸ Downloading all translations from PhraseApp'
+  `phraseapp pull -t #{ARGV[0]}`
+  puts '▸ Removing translations for languages we do not yet support'
+  language_dirs = Dir.glob 'stripe/res/values-*'
+  language_dirs.each do |dir|
+    unless supported_languages.include? dir.split('-')[1]
+      puts 'Ignoring ' + dir + ' because it is not a supported language' if show_info
+      FileUtils.rm_rf(dir) 
+    end
+  end
+
+  supported_languages.each do |lang|
+    puts '==================='
+    puts 'Working on: ' + lang
+    puts '==================='
+    downloaded_filename = 'stripe/res/values-' + lang + '/strings-temp.xml'
+    translations_hash = make_translations_hash(
+      downloaded_filename,
+      included_strings
+    )
+    current_translations_filename = 'stripe/res/values-' + lang + '/strings.xml'
+    remove_excess_strings(
+      current_translations_filename,
+      included_strings,
+      show_info
+    )
+    replace_updated_translations(
+      current_translations_filename, 
+      translations_hash,
+      show_info
+    )
+    add_new_translations(
+      current_translations_filename,
+      translations_hash,
+      show_info
+    )
+    File.delete(downloaded_filename)
+  end
+end
+
+DEFAULT_STRINGS_FILE = 'stripe/res/values/strings.xml'
+@langs = %w(de es fr it ja nl zh)
```

Comment:
Why is this `@langs` as opposed to `langs` or `LANGS`? In the context of a class, `@` would make it an instance variable; but I've never seen an `@`-variable defined outside of a class before. According to [this Stack Overflow result](https://stackoverflow.com/questions/5684066/ruby-creating-instance-variables-outside-a-class) it's defining `@langs` as an instance variable on the "main object". What's going on here?